R/map2.R

```diff
@@ -57,6 +57,13 @@ map2_chr <- function(.x, .y, .f, ..., .progress = FALSE) {
 map2_ <- function(.type, .x, .y, .f, ..., .progress = FALSE, ..error_call = caller_env()) {
   .f <- as_mapper(.f, ...)
   i <- 0L
+
+  .x <- vctrs_list_compat(.x)
```

Comment:
I feel like `map()` should do this just for consistency

Reply:
And in `map_()` you could do:

```r
.x <- vctrs_list_compat(.x)
vec_assert(.x, arg = ".x", call = ..error_call)
```

Since there is no `vec_recycle_common()` call, that would ensure you get the vector check.

This would allow you to remove the usage of `check_vector()` in `map_impl()`, then you can remove that `check_vector()` helper altogether

Reply:
`map_()` probably _has_ to call `vctrs_list_compat()` to ensure that the length is computed correctly when we fix #963